### PR TITLE
Delete post

### PR DESCRIPTION
**Pull request by atwfk**
#242 
- created **DropDown** component with its types, storybook and tests
- created `deletePostHandler` function to delete one post from db and updated posts state
- added `getPostOptions` to handler what options will be passed to the **DropDown** comp based on the user is logged In or not
- added types and tests adjustments to the post view components based on the these updates

### CODE REVIEW

File: modules/shared/components/molecules/PostViewHeader/PostViewHeader.tsx [hunk 1]
```diff
@@ -1,19 +1,48 @@
...
   name,
+  userId,
   date,
-  handlePostOptionsIconClick,
-  id,
+  deletePostHandler,
+  postId,
   isHidden,
 }): ReactElement => {
+  const onOptionClickHandler = (id: string): void => {
```
Comment (omarawad0) [inline modules/shared/components/molecules/PostViewHeader/PostViewHeader.tsx:30]: rename this as well please, **onMenuOptionClickHandler**

File: modules/shared/components/molecules/PostViewHeader/PostViewHeader.tsx [hunk 2]
```diff
@@ -1,19 +1,48 @@
...
 import type { ReactElement, FC } from 'react';
 import Divider from '../../atoms/Divider/Divider';
-import VerticalThreeDotsIcon from '../../icons/verticalThreeDots.svg';
 import type { IPostViewHeader } from './IPostViewHeader';
 import * as EDivider from '../../atoms/Divider/types/EDivider';
 import UserInfo from '../UserInfo/UserInfo';
+import DropDown from '../../atoms/DropDown/DropDown';
+import { options } from '../../atoms/DropDown/mockedOptions';
+
+const getPostOptions = (
```
Comment (omarawad0) [inline modules/shared/components/molecules/PostViewHeader/PostViewHeader.tsx:10]: please change the name to **getPostMenuOptions** or a more specific descriptive name.

**Review by omarawad0 — CHANGES_REQUESTED** [review]
Please consider these required changes:
- the dropDown menu should be closed when the user clicks on either of :
   -  the menu options.
   - anything on the screen other than the menu options.

**Review by ahmedkhitaby — APPROVED** [review]
LGTM

**Review by omarawad0 — APPROVED** [review]

**Pull request merged**